IHP/QueryBuilder.hs

```diff
@@ -753,6 +770,31 @@ innerJoin (name, name') queryBuilderProvider = injectQueryBuilder $ JoinQueryBui
         rightJoinColumn = (Text.encodeUtf8 . fieldNameToColumnName) (symbolToText @name')
 {-# INLINE innerJoin #-}
 
+-- | Index the values from a table with values of a field from a table joined by 'innerJoin' or 'innerJoinThirdTable'. Useful to get, e.g., the comments to a set of posts in such a way that the assignment of comments to posts.
+--
+--
+-- __Example:__ Fetch a list of all comments, each paired with the id of the post it belongs to.
+-- > postLabeledComments <- query @Comment
+-- >                            |> innerJoin @Post (#postId, #id)
+-- >                            |> labelResults @Post #id
+-- >                            |> fetch
+-- > -- SELECT posts.id, comments.* FROM comments INNER JOIN posts ON comments.postId = posts.id 
```

Comment:
I think this example is not good as it could easily be replaced with:

```haskell
comments <- query @Comment
    |> fetch
```

As every comment has. `postId` field, you can just use `get #postId` instead of `get #labelValue`